src/H5Fprivate.h

```diff
@@ -24,11 +24,9 @@ typedef struct H5F_t H5F_t;
 /* Include package's public header */
 #include "H5Fpublic.h"
 
-/* Public headers needed by this file */
-#include "H5FDpublic.h" /* File drivers                 */
-
 /* Private headers needed by this file */
 #include "H5MMprivate.h" /* Memory management            */
+#include "H5FDprivate.h" /* File drivers                 */
```

Comment:
Since H5F_shared_get_file_handles needs the definition for the H5FD_t type, and since H5FDprivate.h includes H5FDpublic.h, just include the private header rather than both here.